earth2mip/diagnostic/base.py

```diff
@@ -0,0 +1,53 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import torch
+from typing import Type, Optional
+from pydantic import BaseModel
+from earth2mip.geo_operator import GeoOperator
+from earth2mip.model_registry import Package
+
+
+class DiagnosticBase(torch.nn.Module, GeoOperator):
+    """Diagnostic model base class"""
+
+    @classmethod
+    def load_package(cls, *args, **kwargs) -> Package:
+        """Class function used to create the diagnostic model package (if needed).
+        This should be where any explicit download functions should be orcastrated
+        """
+        pass
+
+    @classmethod
+    def load_diagnostic(cls, package: Optional[Package], *args, **kwargs):
```

Comment:
diagnostic is already n the class name, so this is redundant

```suggestion
    def load(cls, package: Optional[Package], *args, **kwargs):
```